Cargo.toml

```diff
@@ -18,6 +18,9 @@ maplit = "1.0.2"
 proptest = "1"
 tempfile = "3.3.0"
 
+[dependencies]
+candid = "0.8.4"
```

Comment:
It has been an intentional design decision to not introduce dependencies to this library unless absolutely necessary. In this case, I'm not sure if forcing all users to now depend on candid is the right call. I think we have three options here:

1. Not include this dependency, in which case, users that want to store principals will need to wrap it by defining a `struct WrapperPrincipal(Principal)` in their own crate.
2. We include this dependency, making all users now depend on candid.
3. Add this as an optional feature, so that it can imported by users that need it.

It seems like approach #3 here would be most appropriate. WDYT? cc @dsarlis @roman-kashitsyn

Reply:
Normally I would prefer the approach #3 as well. However I would imagine most people would need to use principals in stable memory, and almost everyone developing on the IC is already dependent on candid in any case.

Reply:
I tend to lean towards option 3 because not everyone might want to use candid to encode what they store in stable memory (there are already cases where people opt for something different because of better performance or more flexible representation for stable data). Having it behind a feature still makes it available relatively easily for the people that want to use candid.

Reply:
I think @0kenx brought up fair arguments though. It's virtually impossible to not depend on candid when building a Rust canister, unless you build a Rust canister that doesn't expose a candid interface, and it's safe to say that these are very, very rare.

The candid dependency isn't related to how people choose to encode/decode their stable data - it's to implement `Storable` for the `Principal` type. I'm starting to see some appeal behind option 1 - most people will need this, given how primitive the `Principal` type is to the IC. In the future, it'd be nice if we move the `Principal` type into a separate crate so that we don't have to depend on the entire candid library.

Reply:
My opinion was not strong, indeed the argument about having a candid interface is a good one. I'm fine with including the dependency too.

Reply:
FYI I'm using a custom newtype in my application now. I'm just saying that if we agree that `principal` and ICRC-1 `(principal, subaccount)` are first-class citizens in the IC, then they should have first-class library support.

Reply:
@0kenx I think that's reasonable. Let's go ahead with that (i.e. stick with option 1 as is done in this PR).